sillsdev/TheCombine
# Get K3s to work without ethernet connection

Fixed some network configuration issues that were preventing k3s/microk8s from starting when the NUC was started up with no Ethernet connection.
 - Fixed the configuration that set the Ethernet connection to `optional`;
 - Create a virtual network interface when the k8s_engine is not `none`. See https://microk8s.io/docs/install-alternatives#heading--offline. Although this is written for `microk8s`, it applies for `k3s` as well.

In addition, this PR:
 - fixes the test that the WiFi Access Point is up after installation;
 - adds an optional playbook to install some packages that can be useful for developers.

Closes #1458

**Verification**
These changes were verified by:
1. Re-installing Ubuntu 20.04.3 Server on a NUC (erasing the old installation)
2. Run `ansible-playbook playbook_dev_tools.yml --limit nuc2 -K` (obviates the need for `-K` later)
3. Run `ansible-playbook playbook_kube_install.yml --limit nuc2`
4. Run `ansible-playbook playbook_kube_config.yml --limit nuc2`; entered `0.7.8` when prompted for the Combine version
5. Set an Android tablet to use the WiFi from `nuc2_ap`
6. Opened Chrome and navigated to `nuc2.thecombine.app`
7. Verified that:
   1. the Combine front page is displayed;
   2. the browser has a valid, secure connection to the site
8. Shutdown the NUC
9. Disconnected the wired ethernet connection
10. Started the NUC
11. Cleared the cache for Chrome on the Android tablet
12. Set the tablet WiFi to `nuc2_ap`
13. Verified that:
   1. the Combine front page is displayed;
   2. the browser has a valid, secure connection to the site;
   3. could register a new user; and
   4. could update the new user's settings


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/sillsdev/thecombine/1455)
